### PR TITLE
Scheduled cron GH-action workflow for updating headers.

### PR DESCRIPTION
**Pull request by Yashwants19**

### CODE REVIEW

File: .github/workflows/update-catch.yaml [hunk 1]
```diff
@@ -0,0 +1,53 @@
+name: Update Catch
+on:
+  pull_request:
+  workflow_dispatch:
+  schedule:
+    - cron:  '0 10 1/16 * *'
```
Comment (Yashwants19) [inline .github/workflows/update-catch.yaml:6]: Hi @zoq and @rcurtin, I guess this is working fine, In the next commit I will remove `pull_request`. This cron will run the workflow at 10:00 UTC every month's 1th and 17th, we can vary this if wanted. We can also tigger this workflow according our use with the help of `workflow_dispatch`.

**Comment by jeffin143** [comment]
you are toooooo fast :-) Usain Bolt

**Comment by Yashwants19** [comment]
#2691 and #2692 was created by github actions bot, which is now closed.

File: .github/workflows/update-catch.yaml (hunk 1 above)
Comment (zoq) [reply .github/workflows/update-catch.yaml:5]: The schedule approach sounds good to me.

File: .github/workflows/update-catch.yaml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+      - name: Update Catch
+        if: steps.catch-header.outputs.current_tag != steps.catch-header.outputs.release_tag
+        env:
+          CURRENT_TAG: ${{ steps.catch-header.outputs.current_tag }}
+          RELEASE_TAG: ${{ steps.catch-header.outputs.release_tag }}
+        run: |
+          # Delete the catch.hpp.
+          rm -f src/mlpack/tests/catch.hpp
+          # Download the release
```
Comment (zoq) [inline .github/workflows/update-catch.yaml:35]:
```suggestion
          # Download the release.
```

File: .github/workflows/update-catch.yaml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    - cron:  '0 10 1/16 * *'
+jobs:
+  updateCatch:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Get Latest Catch Tagged Release
+        id: catch-header
+        run: |
+          # Ping version information upstream
```
Comment (zoq) [inline .github/workflows/update-catch.yaml:14]:
```suggestion
          # Ping version information upstream.
```

Just for consistency.

File: .github/workflows/update-catch.yaml [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+          CATCH_RELEASE_JSON=$(curl -sL https://api.github.com/repos/catchorg/Catch2/releases/latest)
+          CATCH_RELEASE_VERSION=$(jq -r ".tag_name" <<< "$CATCH_RELEASE_JSON" | tr -d v)
+          echo ::set-output name=release_tag::$(echo $CATCH_RELEASE_VERSION)
+          # Extract out version information from git repository
```
Comment (zoq) [inline .github/workflows/update-catch.yaml:18]:
```suggestion
          # Extract out version information from git repository.
```

File: .github/workflows/update-catch.yaml [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+          CATCH_VERSION_MAJOR=$(grep -i ".*#define CATCH_VERSION_MAJOR.*" src/mlpack/tests/catch.hpp | grep -o "[0-9]*")
+          CATCH_VERSION_MINOR=$(grep -i ".*#define CATCH_VERSION_MINOR.*" src/mlpack/tests/catch.hpp | grep -o "[0-9]*")
+          CATCH_VERSION_PATCH=$(grep -i ".*#define CATCH_VERSION_PATCH.*" src/mlpack/tests/catch.hpp | grep -o "[0-9]*")
+          # Combine values to match release tag information
```
Comment (zoq) [inline .github/workflows/update-catch.yaml:22]:
```suggestion
          # Combine values to match release tag information.
```

File: .github/workflows/update-catch.yaml [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+          CATCH_VERSION_VALUE=${CATCH_VERSION_MAJOR}.${CATCH_VERSION_MINOR}.${CATCH_VERSION_PATCH}
+          # Set the current release tag
```
Comment (zoq) [inline .github/workflows/update-catch.yaml:24]:
```suggestion
          # Set the current release tag.
```

File: .github/workflows/update-cli11.yaml [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+    - cron:  '0 10 1/16 * *'
+jobs:
+  updateCLI11:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Get Latest CLI11 Tagged Release
+        id: cli11-header
+        run: |
+          # Ping version information upstream
```
Comment (zoq) [inline .github/workflows/update-cli11.yaml:14]:
```suggestion
          # Ping version information upstream.
```

File: .github/workflows/update-cli11.yaml [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+          CLI11_RELEASE_JSON=$(curl -sL https://api.github.com/repos/CLIUtils/CLI11/releases/latest)
+          CLI11_RELEASE_VERSION=$(jq -r ".tag_name" <<< "$CLI11_RELEASE_JSON" | tr -d v)
+          echo ::set-output name=release_tag::$(echo $CLI11_RELEASE_VERSION)
+          # Extract out version information from git repository
```
Comment (zoq) [inline .github/workflows/update-cli11.yaml:18]:
```suggestion
          # Extract out version information from git repository.
```

File: .github/workflows/update-cli11.yaml [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+          CLI11_VERSION_VALUE=$(grep -i ".*#define CLI11_VERSION.*" src/mlpack/bindings/cli/third_party/CLI/CLI11.hpp | grep -Po "(\d+\.)+\d+")
+          # Set the current release tag
```
Comment (zoq) [inline .github/workflows/update-cli11.yaml:20]:
```suggestion
          # Set the current release tag.
```

File: .github/workflows/update-cli11.yaml [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+
+      - name: Update CLI11
+        if: steps.cli11-header.outputs.current_tag != steps.cli11-header.outputs.release_tag
+        env:
+          CURRENT_TAG: ${{ steps.cli11-header.outputs.current_tag }}
+          RELEASE_TAG: ${{ steps.cli11-header.outputs.release_tag }}
+        run: |
+          # Delete the CLI11.hpp.
+          rm -f src/mlpack/bindings/cli/third_party/CLI/CLI11.hpp
+          # Download the release
```
Comment (zoq) [inline .github/workflows/update-cli11.yaml:31]:
```suggestion
          # Download the release.
```

**Review by zoq — APPROVED** [review]
Looks great to me, thanks for putting this together, just left some really minor comments about the style.

File: .github/workflows/update-catch.yaml [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+
+      - name: Create Pull Request For Catch
+        if: steps.catch-header.outputs.current_tag != steps.catch-header.outputs.release_tag
+        uses: peter-evans/create-pull-request@v3
+        with:
+          commit-message: Upgrade Catch to ${{ steps.catch-header.outputs.release_tag }}
+          title: Upgrade Catch to ${{ steps.catch-header.outputs.release_tag }}
+          body: |
+            Updates (catchorg/Catch2)[https://github.com/catchorg/Catch2] to ${{ steps.catch-header.outputs.release_tag }}.
+            Auto-generated by (create-pull-request)[https://github.com/peter-evans/create-pull-request]
```
Comment (Yashwants19) [inline .github/workflows/update-catch.yaml:46]:
```suggestion
            Updates [catchorg/Catch2](https://github.com/catchorg/Catch2) to ${{ steps.catch-header.outputs.release_tag }}.
            Auto-generated by [create-pull-request](https://github.com/peter-evans/create-pull-request).
```

File: .github/workflows/update-cli11.yaml [hunk 12]
```diff
@@ -0,0 +1,44 @@
...
+
+      - name: Create Pull Request For CLI11
+        if: steps.cli11-header.outputs.current_tag != steps.cli11-header.outputs.release_tag
+        uses: peter-evans/create-pull-request@v3
+        with:
+          commit-message: Upgrade CLI11 to ${{ steps.cli11-header.outputs.release_tag }}
+          title: Upgrade CLI11 to ${{ steps.cli11-header.outputs.release_tag }}
+          body: |
+            Updates (CLIUtils/CLI11)[https://github.com/CLIUtils/CLI11] to ${{ steps.cli11-header.outputs.release_tag }}.
+            Auto-generated by (create-pull-request)[https://github.com/peter-evans/create-pull-request]
```
Comment (Yashwants19) [inline .github/workflows/update-cli11.yaml:42]:
```suggestion
            Updates [CLIUtils/CLI11](https://github.com/CLIUtils/CLI11) to ${{ steps.cli11-header.outputs.release_tag }}.
            Auto-generated by [create-pull-request](https://github.com/peter-evans/create-pull-request).
```

**Review by Yashwants19 — COMMENTED** [review]
Specify links correctly.

**Comment by Yashwants19** [comment]
Okay lets merge this, and start a workflow.

**Pull request merged**